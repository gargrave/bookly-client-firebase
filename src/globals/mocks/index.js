import { authorMocks } from './data/authorMocks';
import { bookBucketMocks } from './data/bookBucketMocks';
import { bookMocks } from './data/bookMocks';
import { profileMocks } from './data/profileMocks';
import { userMocks } from './data/userMocks';

export {
  authorMocks,
  bookBucketMocks,
  bookMocks,
  profileMocks,
  userMocks,
};
