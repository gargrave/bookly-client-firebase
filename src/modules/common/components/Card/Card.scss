@import '../../../../styles/_shadows';

$textColor: #6e6969;
$titleColor: #5d5656;

.card {
  @include boxShadowMed;
  background-color: #f9f9f9;
  border: solid 1px #bbb;
  border-radius: 5px;
  margin: 8px auto;
  max-width: 550px;
  padding: 10px 20px;
}

.hoverableCard:hover {
  background: #e4effc;
  cursor: pointer;
}

.header,
.title,
.text {
  margin: 0;
}

.header,
.title {
  color: $titleColor;
  font-weight: bold;
}

.header {
  font-size: 1.4rem;
}

.detailCard {
  padding: 15px;

  .title {
    font-size: 1.3rem;
    line-height: 2.5rem;
  }

  .title + .cardText {
    font-size: 1.1rem;
  }

  .cardText + .cardText {
    margin-top: 8px;
  }
}

.listDetailCard {
  padding: 10px
}
