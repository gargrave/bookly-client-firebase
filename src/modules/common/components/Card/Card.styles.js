import styled from 'react-emotion'
import { colors, shadows } from '../../../../styles/'

export const StyledWrapper = styled('div')`
  ${shadows.medium};

  background-color: ${colors.cardBg};
  border: solid 1px ${colors.cardBorder};
  border-radius: 5px;
  margin: 8px auto;
  max-width: 550px;
  padding: 10px 20px;

  &.hoverable:hover {
    background: ${colors.cardHoveredBg};
    cursor: pointer;
  }
`
