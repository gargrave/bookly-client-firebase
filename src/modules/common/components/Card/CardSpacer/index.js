import CardSpacer from './CardSpacer'

export default CardSpacer
