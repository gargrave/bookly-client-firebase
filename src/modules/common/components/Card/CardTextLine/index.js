import CardTextLine from './CardTextLine'

export default CardTextLine
