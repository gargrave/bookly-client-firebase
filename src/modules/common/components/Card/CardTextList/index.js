import CardTextList from './CardTextList'

export default CardTextList
