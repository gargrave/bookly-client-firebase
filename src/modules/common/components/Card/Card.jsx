// @flow
import React from 'react'
import { any, bool, func } from 'prop-types'
import styled from 'react-emotion'

import { buildClass } from '../../../../utils/cssHelpers'

import CardHeader from './CardHeader'
import CardSpacer from './CardSpacer'
import CardTextLine from './CardTextLine'
import CardTextList from './CardTextList'

import { colors, shadows } from '../../../../styles/'

type Props = {
  children?: any,
  hoverable?: boolean,
  onClick?: Function,
}

const Styled = styled('div')`
  ${shadows.medium};

  background-color: ${colors.cardBg};
  border: solid 1px ${colors.cardBorder};
  border-radius: 5px;
  margin: 8px auto;
  max-width: 550px;
  padding: 10px 20px;
  text-align: center;

  &.hoverable:hover {
    background: ${colors.cardHoveredBg};
    cursor: pointer;
  }
`

const Card = ({ children, hoverable = false, onClick }: Props) => {
  return (
    <Styled className={buildClass({ hoverable })} onClick={onClick}>
      {children}
    </Styled>
  )
}

Card.Header = CardHeader
Card.Spacer = CardSpacer
Card.TextLine = CardTextLine
Card.TextList = CardTextList

Card.propTypes = {
  children: any,
  hoverable: bool,
  onClick: func,
}

export default Card
