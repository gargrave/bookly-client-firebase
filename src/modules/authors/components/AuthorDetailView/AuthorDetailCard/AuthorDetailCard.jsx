// @flow
import React from 'react';
import { func, instanceOf, oneOfType, shape, string } from 'prop-types';
import { format } from 'date-fns';

import type { Author } from '../../../flowtypes';

import Button from '../../../../common/components/Button/Button';
import ButtonRow from '../../../../common/components/ButtonRow/ButtonRow';
import Card from '../../../../common/components/Card/Card';

type Props = {
  author: Author,
  onBackClick: Function,
  onDeleteClick: Function,
  onEditClick: Function,
};

const authorDatesTextList = (author: Author) => {
  const {
    created,
    updated,
  } = author;

  return [
    { title: 'Added', value: format(created, 'MMM. DD, YYYY, HH:mm:ss') },
    { title: 'Updated', value: format(updated, 'MMM. DD, YYYY, HH:mm:ss') },
  ];
};

const AuthorDetailCard = ({
  author,
  onBackClick,
  onDeleteClick,
  onEditClick,
}: Props) => {
  return (
    <Card
      header={`${author.firstName} ${author.lastName}`}>

      <Card.Spacer />
      <Card.TextList textList={authorDatesTextList(author)} />
      <Card.Spacer size='large' />

      <ButtonRow>
        <Button
          onClick={onEditClick}
          position="left"
          text="Edit"
          type="info" />
        <Button
          onClick={onDeleteClick}
          text="Delete"
          type="danger" />
        <Button
          onClick={onBackClick}
          position="right"
          text="Back"
          type="light" />
      </ButtonRow>
    </Card>
  );
};

AuthorDetailCard.propTypes = {
  author: shape({
    id: string,
    created: oneOfType([
      instanceOf(Date),
      string,
    ]),
    updated: oneOfType([
      instanceOf(Date),
      string,
    ]),
    firstName: string,
    lastName: string,
  }),
  onBackClick: func.isRequired,
  onDeleteClick: func.isRequired,
  onEditClick: func.isRequired,
};

export default AuthorDetailCard;
