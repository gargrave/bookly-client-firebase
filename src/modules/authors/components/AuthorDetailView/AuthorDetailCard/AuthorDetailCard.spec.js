import React from 'react';
import { shallow } from 'enzyme';

import { authorMocks } from '../../../../../globals/mocks/';

import Card from '../../../../common/components/Card/Card';

import AuthorDetailCard from './AuthorDetailCard';

const defaultProps = Object.freeze({
  author: Object.create(authorMocks[0]),
  onBackClick: jest.fn(),
  onDeleteClick: jest.fn(),
  onEditClick: jest.fn(),
});

function getComponent(extraProps = {}) {
  const props = Object.assign({}, defaultProps, extraProps);
  return shallow(<AuthorDetailCard {...props} />);
}

describe('AuthorDetailCard', () => {
  let component;

  test('matches the snapshot', () => {
    component = getComponent();
    expect(component).toMatchSnapshot();
  });

  test('renders correctly', () => {
    component = getComponent();
    expect(component.find(Card)).toHaveLength(1);
  });
});
