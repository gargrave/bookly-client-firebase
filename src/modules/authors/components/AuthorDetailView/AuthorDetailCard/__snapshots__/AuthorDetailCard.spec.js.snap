// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorDetailCard matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthorDetailCard
    author={Object {}}
    onBackClick={[Function]}
    onDeleteClick={[Function]}
    onEditClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <CardSpacer />,
        <CardTextList
          textList={
            Array [
              Object {
                "title": "Added",
                "value": "Jun. 30, 2017, 18:16:43",
              },
              Object {
                "title": "Updated",
                "value": "Jun. 30, 2017, 18:16:43",
              },
            ]
          }
        />,
        <CardSpacer
          size="large"
        />,
        <ButtonRow>
          <Button
            onClick={[Function]}
            position="left"
            text="Edit"
            type="info"
          />
          <Button
            onClick={[Function]}
            text="Delete"
            type="danger"
          />
          <Button
            onClick={[Function]}
            position="right"
            text="Back"
            type="light"
          />
        </ButtonRow>,
      ],
      "classes": Array [
        "detail-card",
        "author-detail-card",
      ],
      "header": "Aldous Huxley",
      "hoverable": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "textList": Array [
            Object {
              "title": "Added",
              "value": "Jun. 30, 2017, 18:16:43",
            },
            Object {
              "title": "Updated",
              "value": "Jun. 30, 2017, 18:16:43",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "size": "large",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Button
              onClick={[Function]}
              position="left"
              text="Edit"
              type="info"
            />,
            <Button
              onClick={[Function]}
              text="Delete"
              type="danger"
            />,
            <Button
              onClick={[Function]}
              position="right"
              text="Back"
              type="light"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "position": "left",
              "text": "Edit",
              "type": "info",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "text": "Delete",
              "type": "danger",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "position": "right",
              "text": "Back",
              "type": "light",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CardSpacer />,
          <CardTextList
            textList={
              Array [
                Object {
                  "title": "Added",
                  "value": "Jun. 30, 2017, 18:16:43",
                },
                Object {
                  "title": "Updated",
                  "value": "Jun. 30, 2017, 18:16:43",
                },
              ]
            }
          />,
          <CardSpacer
            size="large"
          />,
          <ButtonRow>
            <Button
              onClick={[Function]}
              position="left"
              text="Edit"
              type="info"
            />
            <Button
              onClick={[Function]}
              text="Delete"
              type="danger"
            />
            <Button
              onClick={[Function]}
              position="right"
              text="Back"
              type="light"
            />
          </ButtonRow>,
        ],
        "classes": Array [
          "detail-card",
          "author-detail-card",
        ],
        "header": "Aldous Huxley",
        "hoverable": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "textList": Array [
              Object {
                "title": "Added",
                "value": "Jun. 30, 2017, 18:16:43",
              },
              Object {
                "title": "Updated",
                "value": "Jun. 30, 2017, 18:16:43",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": "large",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Button
                onClick={[Function]}
                position="left"
                text="Edit"
                type="info"
              />,
              <Button
                onClick={[Function]}
                text="Delete"
                type="danger"
              />,
              <Button
                onClick={[Function]}
                position="right"
                text="Back"
                type="light"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "position": "left",
                "text": "Edit",
                "type": "info",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "text": "Delete",
                "type": "danger",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "position": "right",
                "text": "Back",
                "type": "light",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
